Sources/OpenAPIRuntime/Conversion/Converter+Server.swift

```diff
@@ -56,11 +56,42 @@ extension Converter {
                 // Drop everything after the optional semicolon (q, extensions, ...)
                 value.split(separator: ";")[0].trimmingCharacters(in: .whitespacesAndNewlines).lowercased()
             }
-
         if acceptValues.isEmpty { return }
-        if acceptValues.contains("*/*") { return }
-        if acceptValues.contains("\(substring.split(separator: "/")[0].lowercased())/*") { return }
-        if acceptValues.contains(where: { $0.localizedCaseInsensitiveContains(substring) }) { return }
+        guard let parsedSubstring = OpenAPIMIMEType(substring) else {
+            throw RuntimeError.invalidAcceptSubstring(substring)
+        }
+        // Look for the first match.
+        for acceptValue in acceptValues {
+            // Fast path.
+            if acceptValue == substring { return }
+            guard let parsedAcceptValue = OpenAPIMIMEType(acceptValue) else {
+                throw RuntimeError.invalidExpectedContentType(acceptValue)
+            }
+            switch (parsedAcceptValue.kind, parsedSubstring.kind) {
+            case (.any, _):
+                // Accept: */* always matches
+                return
+            case (.anySubtype(type: let acceptType), let substring):
+                switch substring {
+                case .any:
+                    // */* as a concrete content type is NOT a match for an Accept header of foo/*
+                    break
+                case .anySubtype(type: let substringType):
+                    // Only match if the types match.
+                    if substringType.lowercased() == acceptType.lowercased() { return }
+                case .concrete(type: let substringType, _):
+                    if substringType.lowercased() == acceptType.lowercased() { return }
+                }
+            case (.concrete(type: let acceptType, subtype: let acceptSubtype), let substring):
+                if case let .concrete(substringType, substringSubtype) = substring {
+                    if acceptType.lowercased() == substringType.lowercased()
+                        && acceptSubtype.lowercased() == substringSubtype.lowercased()
+                    {
+                        return
+                    }
+                }
+            }
+        }
```

Comment:
Functionally I've convinced myself this is fine but I found the combination of doing all accept header values in the loop, the use of nested switch and the early-return to be a challenge to understand.

I'd really like to factor this out into a function that operates on two parsed `OpenAPIMIMETypes` and returns a boolean whether one can satisfy the other. 

Essentially, adding _this_ to `OpenAPIMIMEType`:

```swift
    func satisfies(parsedAcceptValue: OpenAPIMIMEType) -> Bool {
        switch (parsedAcceptValue.kind, self.kind) {
        case (.concrete, .any), (.concrete, .anySubtype), (.anySubtype, .any):
            // The response content-type must be at least as specific as the accept header.
            return false
        case (.any, _):
            // Accept: */* -- Any content-type satisfies the accept header.
            return true
        case (.anySubtype(let acceptType), .anySubtype(let substringType)),
            (.anySubtype(let acceptType), .concrete(let substringType, _)):
            // Accept: type/* -- The content-type should match the partially-specified accept header.
            return acceptType.lowercased() == substringType.lowercased()
        case (.concrete(let acceptType, let acceptSubtype), .concrete(let substringType, let substringSubtype)):
            // Accept: type/subtype -- The content-type should match the concrete type.
            return acceptType.lowercased() == substringType.lowercased() && acceptSubtype.lowercased() == substringSubtype.lowercased()
        }
    }
```

Then this function that does validation and throws can become:

```swift
...
        // Look for the first match.
        for acceptValue in acceptValues {
            // Fast path.
            if acceptValue == substring { return }
            guard let parsedAcceptValue = OpenAPIMIMEType(acceptValue) else {
                throw RuntimeError.invalidExpectedContentType(acceptValue)
            }
            if parsedSubstring.satisfies(parsedAcceptValue: parsedAcceptValue) { return }
        }
        throw RuntimeError.unexpectedAcceptHeader(acceptHeader)

```

It affords for a more testable, composable function too.

Reply:
Thanks, updated!